### PR TITLE
Add devise gem to setup for signup and login

### PR DESCRIPTION
**Pull request by AntonellaMorittu**
#### Description
Changes proposed in this pull request:

*  Devise setup
*  Signup feature 
*  Login feature

#### What I have learned working on this feature: 
How to setup devise in my app and test the signup and login feature.

### CODE REVIEW

File: features/signup.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+   Scenario: User write wrong password
+     Given User visits the site
+     And User clicks on "Sign up" button
+     When I fill "Password" with "123"
+     And I click on "Sign up"
+     Then I should see message "Password is too short (minimum is 6 characters)"
+   #
+   #
+   #    Password is too short (minimum is 8 characters)
```
Comment (tochman) [inline features/signup.feature]: Please remove comments

File: features/step_definitions/login_steps.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Given("following user exist") do |table|
+ table.hashes.each do |user|
+   FactoryBot.create(:user, user)
+ end
+end
+
+Given("I am on the landing-page") do
```
Comment (tochman) [inline features/step_definitions/login_steps.rb]: This language suggests that this an assertion but it’s not

File: features/step_definitions/login_steps.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  visit root_path
+end
+
+Given("I click {string}") do |button|
+  click_on button
+end
+
+When("I fill in {string} with {string}") do |input, value|
+fill_in(input, :with => value)
```
Comment (tochman) [inline features/step_definitions/login_steps.rb]: Format code, use poetry mode, remove hash rocket

File: features/step_definitions/signup_steps.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Given("User visits the site") do
+  visit root_path
+end
+
+Then("User clicks on {string} button") do |button|
+  click_on(button)
+end
```
Comment (tochman) [inline features/step_definitions/signup_steps.rb]: This is code repetition. See a step definition in this very file with the same capybara command

File: features/step_definitions/signup_steps.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Then("I fill {string} with {string}") do |input, value|
+  fill_in(input, :with => value)
+end
```
Comment (tochman) [inline features/step_definitions/signup_steps.rb]: Another code repetition

File: features/step_definitions/signup_steps.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+Then("I click on {string}") do |button|
+  click_on(button)
```
Comment (tochman) [inline features/step_definitions/signup_steps.rb]: Code repetition??

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/login_steps.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+Given("I click {string}") do |button|
+  click_on button
+end
+
+When("I fill in {string} with {string}") do |input, value|
+  fill_in input, with: value
+end
+
+When("I click {string} button") do |button|
+  click_on(button)
```
Comment (magnus-thor) [inline features/step_definitions/login_steps.rb]: You have three steps that do the same, click a button, in you feature file word them all in the same way and then you only need 1 step for them all.

File: features/application.feature [hunk 8]
```diff
@@ -1,4 +1,4 @@
-Feature: User visits the home page
+Feature: I visit the site the home page
```
Comment (magnus-thor) [inline features/application.feature]: change `I visit the site the home page` to `I visit the site`

Comment (NellaMori) [reply features/application.feature]: @magnus-thor fixed! 😄

**Review by tochman — APPROVED** [review]

**Pull request merged**